tests/migration_test.go

```diff
@@ -2946,6 +2946,99 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				expectFeatureToBeSupportedOnNode(newNode, requiredFeatures)
 			})
 
+			Context("[Serial]node with obsolete host-model cpuModel", func() {
+
+				var node *k8sv1.Node
+				var obsoleteModel string
+				var kvConfig *v1.KubeVirtConfiguration
+
+				BeforeEach(func() {
+					node = &(libnode.GetAllSchedulableNodes(virtClient).Items[0])
+					obsoleteModel = tests.GetNodeHostModel(node)
+
+					By("Updating Kubevirt CR , this should wake node-labeller ")
+					kvConfig = util.GetCurrentKv(virtClient).Spec.Configuration.DeepCopy()
+					if kvConfig.ObsoleteCPUModels == nil {
+						kvConfig.ObsoleteCPUModels = make(map[string]bool)
+					}
+					kvConfig.ObsoleteCPUModels[obsoleteModel] = true
+					tests.UpdateKubeVirtConfigValueAndWait(*kvConfig)
+
+					Eventually(func() error {
+						node, err = virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+						Expect(err).ShouldNot(HaveOccurred())
+
+						_, exists := node.Annotations[v1.LabellerSkipNodeAnnotation]
+						if exists {
+							return fmt.Errorf("node %s is expected to not have annotation %s", node.Name, v1.LabellerSkipNodeAnnotation)
+						}
+
+						obsoleteModelLabelFound := false
+						for labelKey, _ := range node.Labels {
+							if strings.Contains(labelKey, v1.NodeHostModelIsObsoleteLabel) {
+								obsoleteModelLabelFound = true
+								break
+							}
+						}
+						if !obsoleteModelLabelFound {
+							return fmt.Errorf("node %s is expected to have a label with %s substring. this means node-labeller is not enabled for the node", node.Name, v1.NodeHostModelIsObsoleteLabel)
+						}
+
+						return nil
+					}, 30*time.Second, time.Second).ShouldNot(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					delete(kvConfig.ObsoleteCPUModels, obsoleteModel)
+					tests.UpdateKubeVirtConfigValueAndWait(*kvConfig)
```

Comment:
> In here i delete only the cpuModel of the node from ObsoleteCPUModels,
> i added it my self to this list so the original list should stay the same.
> (Originally the node's cpuModel should never be there so there is no need to check it)

You're right! My bad

> That's a good point TestCleanup() does restore the default configuration but
> it takes some time until the node labeler start working so we should make sure
> it updated the labels BEFORE THE NEXT TEST STARTS this is important to prevent
> flakiness.

Got you